### ISSUE TITLE
[International Draughts] PDN support enhancements

### ISSUE THREAD

**Issue by IgorLeMasson** [mi-g/joclyboard#11]
Initial wishlist for PDN support enhancements:

**New tags**
1.1. Add support to others ['mandatory' tags: **Event**, **Site** and **Round**](http://pdn.fmjd.org/pdntags.html#mandatory-tags). User should have a option to add them.

1.2. Add support to [**GameType**](http://pdn.fmjd.org/gametype.html#gametype-section) tag. Currently in PJN file, the game type is written in a non-PDN tag "JoclyGame": **[JoclyGame "draughts"]**.

1.3. Add support to [**TimeControl**](http://pdn.fmjd.org/pdntags.html#mandatory-tags) tag, according default or user-defined time settings

**File extension**
2.1. Export to PDN file extension could be added or replace PJN file option.

**PDN Export**
3.1. The export button might be available on the board window botton toolbar, instead of only on **History** window botton toolbar.  On the board window, the "**Save**" button only accept JSON format

**Read/Import PDN files**
4.1. It is not yet supported

**Comment by mi-g** [comment]
1.*, 5.1: those tags cannot be filled automatically by Jocly, this would require specific UI and action from the user to be specified. Given this, the strategy is that the user can as well use a text editor.

2.1: can't you edit the file extension from the SaveAs dialog ? (it might be OS dependent)

3.1: the actual problem is that there are already to many options in the toolbar at the bottom of the Play window. This is the reason why the export feature has been located to the history window.
We should find a way to improve the UI to prevent the list of icons to split the icons over 2 lines. 

4.1: can you share some non-working PDN ? Note that because the feature is supposed to be game-agnostic, some lexical errors in the PDN cannot be auto-corrected without breaking the parsing of other games.

**Comment by IgorLeMasson** [comment]
> 1.*, 5.1: those tags cannot be filled automatically by Jocly, this would require specific UI and action from the user to be specified. Given this, the strategy is that the user can as well use a text editor.

The tags mentioned on 1.1, most of draughts softwares that have a proper PDN support allows user-input fields for these tags. PDN knowledge should be transparent to the users. as you told a GUI form to user-input is the better way to deal it.

GameType and TimeControl tags could be automatically added based on played raughts variant and TimeControl (default and user-defined)

> 2.1: can't you edit the file extension from the SaveAs dialog ? (it might be OS dependent)

Of course it can be done. but now the **PJN** file is not PDN 3.0 compliant since it has a non-PDN tag: **[JoclyGame "draughts"]**, if it haven't been used by Jocly framework, this tag could be removed and filename extension could be changed from PJN to PDN, then the exported file would be **100%** compliant with PDN 3.0.

**Comment by mi-g** [comment]
We have to make compromises between some game-specific needs and keeping the UI simple. If we want a program that is purely for draughts, the best is to change/recompile with the specifics.

> Maybe reducing the number of icons?
We cannot do that, those icons are there for a reason: provide access to existing features.

I tried a few modydam PDNs and all of them loaded and played without a problem. Could it be that you were trying to open the PDN from the _Load_ button ? Instead, you must open the Game window, then button _Open book_.
Note that in those PDN examples, none of the tags you requested are present.

**Comment by IgorLeMasson** [comment]
Yes, on my previous tests I have used _Load_ button. I will test with others PDN files and share the results.

**Comment by mi-g** [comment]
In the main window (the one that lists all available games), if you click a game entry (not the white triangle inside a green disk that launches a match directly), this opens the game window. This is also where you start clocked matches from.

**Comment by IgorLeMasson** [comment]
Ok now I see it. Should be a better way to access this option, keeps it in another window  is not useful rather than board or even history window. The "Open book" name  can be replace to "Open game". 
The term book can mislead refering to "Opening book" concept instead of 'saved game' or 'game database'.

**Comment by mi-g** [comment]
I agree _Open book_ is not good, but i believe _Open game_ is worse. _Open archive_ maybe ?

**Comment by IgorLeMasson** [comment]
Maybe '_Open saved game_' ?!

**Comment by mi-g** [comment]
We should somehow have this notion that a PDN/PGN/PJN is a collection of games not a single one. _Book_ might be the right word, but _Open_ is confusing because it can be misunderstood with _game opening_, the first moves of a game. What about _Load book_ ?

**Comment by IgorLeMasson** [comment]
Hi Michel,
I guess just "Open game(s) file" is enough.

Backing to the original issue, regarding **4.1**,  that maybe deserves a separated issue: please check the attached PDN files. Later to enrich test coverage, I will test others PDN files to try to found unexpected results. 

[PDN_files_to_test.zip](https://github.com/mi-g/joclyboard/files/1100708/PDN_files_to_test.zip)
